content/blog/2021/08/using-maxsat-for-resolving-deps.md

```diff
@@ -0,0 +1,171 @@
+# Hypper: using a MaxSAT for resolving chart dependencies on Kubernetes
+
+[Hypper](http://hypper.io) is a Kubernetes package manager based on Helm. It
+handles charts in a similar way as Helm does, with a twist: it tries to resolve
+dependencies between charts to reuse existing resources and deployments.
+For more info, see the two articles on this blog by @mattfarina: [Introducing
+Hypper](https://community.suse.com/posts/15893197) and
+[Hypper v3 - Now with a SAT solver](https://community.suse.com/posts/14072167).
+
+Hypper is designed to work with shared dependencies: dependencies installed once
+on a cluster that several charts depend on. Think of Hypper as bringing the
+usual Linux package management to Kubernetes Helm charts.
+
+Checking all the charts to see if their dependencies are correctly installed or
+there are some missing is not an easy task. This is called a "Satisfiability
+problem", or SAT, for friends. In fact, solving a SAT problem it's an [NP-Complete
+problem](https://en.wikipedia.org/wiki/Boolean_satisfiability_problem#Unrestricted_satisfiability_(SAT))!.
+
+## How to solve a SAT problem?
+
+Finding a solution to that problem may be somewhat easy by reducing the problem
+space (like Go modules), or using custom heuristics to be fast but without an
```

Comment:
You might want to link to "Go modules". Many people who read this may not know Go or even what Go modules are. I would use the URL https://golang.org/ref/mod.

Reply:
Resolved with 8ceaae9